src/table/styles.ts

```diff
@@ -122,6 +123,46 @@ export const VisuallyHidden = styled('span')({
   width: 1,
 });
 
+export const HeadCellContent = styled(Box)({
+  display: 'flex',
+  flexDirection: 'inherit',
+  justifyContent: 'space-between',
+});
+
+export const StyledMenuIconButton = styled(IconButton)({
+  opacity: 0,
+  minWidth: '30px',
+  padding: '4px',
+  radius: '3px',
+  '&:focus': {
+    opacity: 1,
+  },
+});
+
+export const StyledCellMenu = styled('div', {
+  shouldForwardProp: (prop: string) => prop !== 'headerStyle',
+})(({ headerStyle }) => ({
+  '.MuiPaper-root': {
+    width: '220px',
+  },
+  '.MuiListItemText-primary, .MuiSvgIcon-root': {
+    fontSize: '16px',
+    color: headerStyle.sortLabelColor,
+  },
+  '.head-cell-menu': {
+    color: headerStyle.sortLabelColor,
+  },
+}));
+
+export const StyledTableCell = styled(TableCell, {
+  shouldForwardProp: (prop: string) => prop !== 'headerStyle',
+})(({ headerStyle }) => ({
+  ...headerStyle,
```

Comment:
the headerStyle prop does not exist here, so this one should be static. There are a couple of examples of the syntax for this, e.g. `VisuallyHidden`